pallets/moonbeam-lazy-migrations/src/lib.rs

```diff
@@ -124,5 +130,55 @@ pub mod pallet {
 			}
 			Ok(Pays::No.into())
 		}
+		#[pallet::call_index(2)]
+		#[pallet::weight(Pallet::<T>::create_contract_metadata_weight(MAX_CONTRACT_CODE_SIZE))]
+		pub fn create_contract_metadata(
+			origin: OriginFor<T>,
+			address: H160,
+		) -> DispatchResultWithPostInfo {
+			ensure_signed(origin)?;
+
+			ensure!(
+				pallet_evm::AccountCodesMetadata::<T>::get(address).is_none(),
+				Error::<T>::ContractMetadataAlreadySet
+			);
+
+			// Ensure contract exist
+			let code = pallet_evm::AccountCodes::<T>::get(address);
+			ensure!(!code.is_empty(), Error::<T>::ContractNotExist);
+
+			// Construct metadata
+			let code_size = code.len() as u64;
+			let code_hash = sp_core::H256::from(sp_io::hashing::keccak_256(&code));
+			let meta = pallet_evm::CodeMetadata {
+				size: code_size,
+				hash: code_hash,
+			};
+
+			// Set metadata
+			pallet_evm::AccountCodesMetadata::<T>::insert(address, meta);
+
+			Ok((
+				Some(Self::create_contract_metadata_weight(code_size)),
+				Pays::No,
```

Comment:
Not sure about the `Pays::No` here. Theoretically we can call `create_contract_metadata` for any contract and we are not removing any storage, only adding.

Reply:
This should only be callable for addresses in the assetManager, right

Reply:
It's not about assetManager, it's about old evm contract created before the introduction of pallet evm storage Accout AccountCodesMetadata.
So, it can success only once per contract